pkg/apis/sparkoperator.k8s.io/v1alpha1/types.go

```diff
@@ -238,13 +238,15 @@ type ApplicationStateType string
 
 // Different states an application may have.
 const (
-	NewState              ApplicationStateType = ""
-	SubmittedState        ApplicationStateType = "SUBMITTED"
-	RunningState          ApplicationStateType = "RUNNING"
-	CompletedState        ApplicationStateType = "COMPLETED"
-	FailedState           ApplicationStateType = "FAILED"
-	FailedSubmissionState ApplicationStateType = "SUBMISSION_FAILED"
-	PendingRetryState     ApplicationStateType = "PENDING_RETRY"
+	NewState                 ApplicationStateType = ""
+	SubmittedState           ApplicationStateType = "SUBMITTED"
+	RunningState             ApplicationStateType = "RUNNING"
+	CompletedState           ApplicationStateType = "COMPLETED"
+	FailedState              ApplicationStateType = "FAILED"
+	FailedSubmissionState    ApplicationStateType = "SUBMISSION_FAILED"
+	PendingRetryState        ApplicationStateType = "PENDING_RETRY"
+	TerminatedCompletedState ApplicationStateType = "TERMINATED_COMPLETED"
```

Comment:
What is this state? You have a pending retry state, would that not be enough?
Also i think we should add intermediate states, rather than new terminal states, as this makes it backwards incompatible

Reply:
I think they are equivalent since we will be backward incompatible either way ?